neurodamus/node.py

```diff
@@ -853,10 +854,13 @@ def enable_reports(self):
             if SimConfig.restore_coreneuron:
                 continue  # we dont even need to initialize reports
 
-            report = Nd.Report(*rep_params)
+            has_gids = self._circuits.global_manager.get_final_gids().size > 0
+            report = Report(*rep_params) if has_gids else None
 
             if not SimConfig.use_coreneuron or rep_params.rep_type == "Synapse":
-                self._report_setup(report, rep_conf, target, rep_params.rep_type)
+                if not self._report_setup(report, rep_conf, target, rep_params.rep_type):
+                    n_errors += 1
+                    continue
```

Comment:
Instead of having `_report_setup` return a `bool` and catch any exception, you could do the exception catching here. The one early `return False` could also `raise`.

Would sort of force exception/error treatment in the caller (here), which seems a bit more appropriate, and it makes it harder to forget checking the error return value.

Reply:
the idea was to show the errors in all reports before raising the exception, that way users can fix all problems and not going one by one...

Reply:
I meant something along
```
try:
    self._report_setup(...)
except:
    n_errors += 1
```
to avoid anyone not ever checking the return value of `_report_setup`